src/components/segmentedControl/index.tsx

```diff
@@ -0,0 +1,188 @@
+import _ from 'lodash';
+import React, {useRef, useState} from 'react';
+import {StyleSheet, StyleProp, ViewStyle, LayoutChangeEvent} from 'react-native';
+import Reanimated, {EasingNode, Easing as _Easing} from 'react-native-reanimated';
+import {Colors, BorderRadiuses, Spacings} from '../../style';
+import {asBaseComponent} from '../../commons/new';
+import View from '../view';
+import Segment, {SegmentItemProps} from './segment';
+
+const {interpolate: _interpolate, interpolateNode} = Reanimated;
+const interpolate = interpolateNode || _interpolate;
+const Easing = EasingNode || _Easing;
+const BORDER_WIDTH = 1;
+
+export type SegmentedControlProps = {
+  /**
+   * Array on segments.
+   */
+  segments?: SegmentItemProps[];
+  /**
+   * The color of the active segment (label and outline).
+   */
+  activeColor?: string;
+  /**
+   * The color of the inactive segments (label).
+   */
+  inActiveColor?: string;
+  /**
+   * Callback for when index has change.
+   */
+  onChangeIndex?: (index: number) => void;
+  /**
+   * Initial index to be active.
+   */
+  initialIndex?: number;
+  /**
+   * The segmentedControl borderRadius
+   */
+  borderRadius?: number;
+  /**
+   * The background color of the inactive segments
+   */
+  backgroundColor?: string;
+  /**
+   * The background color of the active segment
+   */
+  activeBackgroundColor?: string;
+  /**
+   * The color of the segmentedControl outline
+   */
+  outlineColor?: string;
```

Comment:
I would think this prop is meant for the active indicator and not the whole segment control. 
For the whole SegmentControl the user can pass the containerStyle. WDYT?